File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int>mpp;
        vector<int>ans;
        for(int i = 0; i < nums.size(); i++){
            mpp[nums[i]]++;
        }
        priority_queue<pair<int, int>>pq;

        for(auto it : mpp){
            pq.push({it.second, it.first});
        }
        while(k > 0){
            pair temp = pq.top();
            ans.push_back(temp.second);
            pq.pop();
            k--;
        }
        return ans;
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        ans = []
        mpp = {}

        for x in nums:
            mpp[x] = mpp.get(x, 0) + 1

        for i in range(k):
            maxi = 0
            temp = -1000000000
            for key,value in mpp.items():
                if(value > maxi):
                    maxi = value
                    temp = key
            ans.append(temp)
            mpp[temp] = -1
        return ans
        


        